src/app/posts/edit/edit.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Post } from '../../post';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/internal/operators';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-edit',
+  templateUrl: './edit.component.html',
+  styleUrls: ['./edit.component.scss']
+})
+export class EditComponent implements OnInit {
+  post: Post;
+  profileForm = null;
+
+  constructor(
+    private httpClient: HttpClient,
+    private activatedRoute: ActivatedRoute,
+    private router: Router,
+    private formBuilder: FormBuilder
+  ) { }
+
+  ngOnInit() {
+    const id = this.activatedRoute.snapshot.params.id;
+
+    this.httpClient.get('https://jsonplaceholder.typicode.com/posts/' + id)
+      .pipe(map((res: any) => {
+        return new Post(res);
+      }))
+      .subscribe((res: Post) => {
+        this.profileForm = this.formBuilder.group({
```

Comment:
this.profileForm should be defined on init, not on subscribe. If you want to set value for specific control you can do that in subscribe after you have the data, but the form and controls should be defined on init. 
`this.controls[name].setValue(value);
this.controls[name]. updateValueAndValidity()`